pymeasure/instruments/nanovna/nanovna.py

```diff
@@ -0,0 +1,255 @@
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from numpy import array as _array
+# from pymeasure.adapters import SerialAdapter
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set
+
+
+class NanoVNA(Instrument):
+    """ Represents the NanoVNA interface for interacting with the instrument.
+
+    .. code-block:: python
+
+    """
+
+    def __init__(self, adapter, name="NanoVNA", **kwargs):
+        # kwargs.setdefault('read_termination', '\r\nch> ')
+        # kwargs.setdefault('write_termination', '\r')
+        self.write_termination = '\r'
+        self.read_termination = '\r\nch> '
+        self.echo_termination = '\r\n'
+        super().__init__(
+            adapter,
+            name,
+            includeSCPI=False,
+            # read_termination='\r\nch> ',
+            # write_termination='\r',
+            **kwargs
+        )
+
+    def read(self):
+        """
+        Reads from the instrument including the correct termination characters
+        """
+        # ret = self.adapter._read_echo()
+        # return ret
+
+        read = self.adapter._read_bytes(-1, break_on_termchar=True).decode()
+        read = read.replace('ch> ', "").split(self.echo_termination)
+        # echo = read[0]
+        reply = read[1:-1]
+
+        return ",".join(reply)  # TODO maybe don't return as a single string???
+
+    def write(self, command):
+        """
+        Writes to the instrument including the device address
+
+        :param command: command string to be sent to the instrument
+        """
+        # super().write(command)
+        self.adapter._write(command + self.write_termination)
+
+    def ask(self, command, query_delay=0):
+        """Write a command to the instrument and return the read response.
+
+        :param command: Command string to be sent to the instrument.
+        :param query_delay: Delay between writing and reading in seconds.
+        :returns: String returned by the device without read_termination.
+        """
+        self.write(command)
+        self.wait_for(query_delay)
+        return self.read()
+
+    help = Instrument.measurement(
```

Comment:
The necessary commands for requesting or controlling should be implemented. You therefore don't need a help regarding the lower layer device commands but just a documentation about your wrapper.